### PR TITLE
Static now consider screen

### PR DESCRIPTION
**Pull request by superpupervlad**
Now static consider screen and possible to call static function without screen, default screen is current screen.
#1863

### CODE REVIEW

File: libqtile/window.py [hunk 1]
```diff
@@ -955,7 +956,7 @@ def toggleminimize(self):
         warnings.warn("toggleminimize is deprecated, use toggle_minimize", DeprecationWarning)
         self.toggle_minimize()
 
-    def cmd_static(self, screen, x=None, y=None, width=None, height=None):
+    def cmd_static(self, screen=None, x=None, y=None, width=None, height=None):
```
Comment (tych0) [inline libqtile/window.py:959]: Is this part of the patch necessary? It seems like it is not.

Comment (superpupervlad) [reply libqtile/window.py:959]: This is not necessary, but without this it will be required to specify screen.

Comment (tych0) [reply libqtile/window.py:959]: Right, but it wasn't before, so is there a reason to relax this requirement?

Comment (superpupervlad) [reply libqtile/window.py:959]: Well, in my view if someone will want to use it(place static window not on a certain screen, but on current screen) he have to write extra lines of code, on other if someone want to use specific screen he already specify it.

Comment (tych0) [reply libqtile/window.py:959]: Yep, fair point. So this looks fine, can you add a test for the whole thing?

Comment (superpupervlad) [reply libqtile/window.py:959]: Um, I don't think I can handle it :disappointed_relieved:

File: libqtile/window.py [hunk 2]
```diff
@@ -715,13 +715,14 @@ def __init__(self, win, qtile, screen,
         self.conf_y = y
         self.conf_width = width
         self.conf_height = height
-        self.x = x or 0
-        self.y = y or 0
+        x = x or 0
+        y = y or 0
+        self.x = x + screen.cmd_info()['x']
+        self.y = y + screen.cmd_info()['y']
```
Comment (ramnes) [inline libqtile/window.py:721]: Can't we get the screen position in a simpler way?

File: libqtile/window.py [hunk 3]
```diff
@@ -715,13 +715,14 @@ def __init__(self, win, qtile, screen,
         self.conf_y = y
         self.conf_width = width
         self.conf_height = height
-        self.x = x or 0
-        self.y = y or 0
+        x = x or 0
+        y = y or 0
+        self.x = x + screen.cmd_info()['x']
```
Comment (superpupervlad) [inline libqtile/window.py:720]:
```suggestion
        self.x = x + screen.x
```

File: libqtile/window.py (hunk 2 above)
Comment (superpupervlad) [reply libqtile/window.py:721]: Of course
```suggestion
        self.y = y + screen.y
```
Also maybe to get rid of this checks for None
```
        self.x = x or 0
        self.y = y or 0
```
we should change default values of x and y in line 711 to
`                 x=0, y=0, width=None, height=None):`

**Comment by m-col** [comment]
re testing: this patch should probably do it:

```
diff --git a/test/test_manager.py b/test/test_manager.py
index 41d1c42a..cfaa49d1 100644
--- a/test/test_manager.py
+++ b/test/test_manager.py
@@ -530,7 +530,9 @@ def test_static(qtile):
 
     self.test_xeyes()
     self.test_window("one")
-    self.c.window[self.c.window.info()["id"]].static(0, 0, 0, 100, 100)
+    self.c.window[self.c.window.info()["id"]].static(width=100, height=100)
+    self.test_window("two")
+    self.c.window[self.c.window.info()["id"]].static(screen=0, x=1, y=1)
 
 
 @manager_config
```

**Comment by m-col** [comment]
re testing: this patch should probably do it:

```
diff --git a/test/test_manager.py b/test/test_manager.py
index 41d1c42a..cfaa49d1 100644
--- a/test/test_manager.py
+++ b/test/test_manager.py
@@ -530,7 +530,9 @@ def test_static(qtile):
 
     self.test_xeyes()
     self.test_window("one")
-    self.c.window[self.c.window.info()["id"]].static(0, 0, 0, 100, 100)
+    self.c.window[self.c.window.info()["id"]].static(width=100, height=100)
+    self.test_window("two")
+    self.c.window[self.c.window.info()["id"]].static(screen=0, x=1, y=1)
 
 
 @manager_config
```

**Comment by m-col** [comment]
Hmm actually that isn't enough. We should stick a static window onto screen number 2 and test it's geometry.

**Comment by m-col** [comment]
Hmm actually that isn't enough. We should stick a static window onto screen number 2 and test its geometry.

File: libqtile/window.py (hunk 2 above)
Comment (m-col) [reply libqtile/window.py:721]: Yeah that sounds good

Comment (m-col) [reply libqtile/window.py:721]: After this change is made, I can merge it and try again to make a test. IMO a squashed bug without a test is better than a present bug.

**Comment by ramnes** [comment]
It looks like I found a bug on GitHub as it closed the PR right after I rebased the branch. :(

Merged manually in f271f13f, thanks!

**Comment by m-col** [comment]
Awesome thanks. I pinky swear to put a test on my To Dos 😉

**Pull request closed without merging**